homeassistant/components/hausbus/event_handler.py

```diff
@@ -0,0 +1,20 @@
+"""Event handler abstract class."""
+
+from abc import ABC, abstractmethod
+from collections.abc import Callable, Coroutine
+from typing import Any
+
+from .entity import HausbusEntity
+
+
+class IEventHandler(ABC):
```

Comment:
This is not Java, move this to `gateway.py`

Reply:
This will also create circular dependencies. I moved it outside of gateway.py to define the interface between the gateway and the platform entities.

Reply:
How can there be a circular dependency when the class is only referenced in `gateway.py`?

Reply:
You are right. After the gateway instance moved to the HausbusConfig entry the circular dependency was already solved. This also means that the IEventHandler class is not even necessary and I will completely remove it.